tasks/module_4/task_2.py

```diff
@@ -0,0 +1,51 @@
+"""Module4/task2: File Extension Handler."""
+import os
+from contextlib import contextmanager
+
+
+FILE_PREFIX = 'spam'
+FILES_DIR = 'resources/task_2'
+
+
+@contextmanager
+def cwd(new_dir):
+    """Custom-made context manager for temporarily directory switching during runtime."""
+    old_dir = os.getcwd()
+    os.chdir(new_dir)
+    try:
+        yield
+    finally:
+        os.chdir(old_dir)
+
+
+def get_files_with_prefix() -> list:
+    """Get files with a specified prefix.
+
+    :return list of files matching a given prefix.
+    """
+    matching_files = []
+    for file_ in os.listdir():
+        if os.path.isfile(file_) and file_.startswith(FILE_PREFIX):
+            matching_files.append(file_)
+    return matching_files
+
+
+def adjust_files(files: list):
+    """Rename file to match ordering pattern.
+
+    :param files: list of files to adjust.
+    """
+    extensions = set(os.path.splitext(file_)[1] for file_ in files)
+    for extension in extensions:
```

Comment:
@pdamec could you explain please why do you need nested for loops? Wouldn't it be possible to iterate only over sorted list of files?